### PR TITLE
[Model Monitor] - Update Evaluate Threshold signature to OutputV2

### PR DESCRIPTION
**Pull request by mastloui-msft**
Change evaluate threshold to leverage "group" rather than feature_name & data_type.
Change all signals to also output both group and group_pivot as part of their output.

### CODE REVIEW

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py [hunk 1]
```diff
@@ -99,12 +105,16 @@ def compute_ndcg_and_write_to_mltable(baseline_explanations, production_explanat
...
             constants.METRIC_NAME_COLUMN: "BaselineFeatureImportance",
+            constants.GROUP_COLUMN: production_feature[constants.FEATURE_CATEGORY_COLUMN],
+            constants.GROUP_PIVOT_COLUMN: "",
             constants.THRESHOLD_VALUE: float("nan")}
         production_feature_importance_data = {
             constants.FEATURE_NAME_COLUMN: production_feature[constants.FEATURE_COLUMN],
             constants.METRIC_VALUE_COLUMN: production_feature[constants.METRIC_VALUE_COLUMN],
             constants.FEATURE_CATEGORY_COLUMN: production_feature[constants.FEATURE_CATEGORY_COLUMN],
             constants.METRIC_NAME_COLUMN: "ProductionFeatureImportance",
+            constants.GROUP_COLUMN: production_feature[constants.FEATURE_CATEGORY_COLUMN],
```
Comment (hawestra) [inline assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py:116]: should this be constant.GROUP_COLUMN: production_feature[constants.GROUP_COLUMN] ?

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py [hunk 2]
```diff
@@ -99,12 +105,16 @@ def compute_ndcg_and_write_to_mltable(baseline_explanations, production_explanat
             constants.METRIC_VALUE_COLUMN: baseline_feature[constants.METRIC_VALUE_COLUMN],
             constants.FEATURE_CATEGORY_COLUMN: baseline_feature[constants.FEATURE_CATEGORY_COLUMN],
             constants.METRIC_NAME_COLUMN: "BaselineFeatureImportance",
+            constants.GROUP_COLUMN: production_feature[constants.FEATURE_CATEGORY_COLUMN],
```
Comment (hawestra) [inline assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py:108]: Same as below?

File: assets/model_monitoring/components/data_drift/data_drift_signal_monitor/spec.yaml
Comment (chupadhy) [inline assets/model_monitoring/components/data_drift/data_drift_signal_monitor/spec.yaml:1]: The evaluate metric threshold version is not updated

File: assets/model_monitoring/components/prediction_drift/prediction_drift_signal_monitor/spec.yaml
Comment (chupadhy) [inline assets/model_monitoring/components/prediction_drift/prediction_drift_signal_monitor/spec.yaml:1]: Compute metrics version is not updated

File: assets/model_monitoring/components/data_drift/data_drift_signal_monitor/spec.yaml
Comment (mastloui-msft) [reply assets/model_monitoring/components/data_drift/data_drift_signal_monitor/spec.yaml:1]: :O

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py (hunk 1 above)
Comment (mastloui-msft) [reply assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_attribution_drift.py]: the group for the feature importance is the feature! It's feature importance GROUP BY feature!

**Review by chupadhy — APPROVED** [review]

**Pull request merged**